driver-core/src/main/java/com/datastax/driver/core/policies/DefaultRetryPolicy.java

```diff
@@ -137,6 +138,18 @@ public RetryDecision onUnavailable(Statement statement, ConsistencyLevel cl, int
             : RetryDecision.rethrow();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation triggers a retry on the next host in the query plan,
+     * regardless of the consistency level, the number of retries or
+     * the possibility that the mutation has been applied server-side.
+     */
+    @Override
+    public RetryDecision onRequestError(Statement statement, ConsistencyLevel cl, DriverException e, int nbRetry) {
```

Comment:
There is somewhat of a contradiction with this behavior and the RetryPolicy#onRequestError javadoc:

```java
+     * Note that when this method is invoked, <em>the driver cannot guarantee that the mutation has
+     * been effectively applied server-side</em>; a retry should only be attempted if the request
+     * is known to be idempotent.
```

Here we are always retrying regardless of whether or not the query is idempotent, shouldn't the default behavior match that of the parent java doc?